streams/test-utils/src/test/java/org/apache/kafka/streams/TopologyTestDriverTest.java

```diff
@@ -1088,4 +1090,66 @@ public void shouldCreateStateDirectoryForStatefulTopology() {
         final TaskId taskId = new TaskId(0, 0);
         assertTrue(new File(appDir, taskId.toString()).exists());
     }
+
+    @Test
+    public void shouldReturnIteratorForMultipleRecords() {
+        testDriver = new TopologyTestDriver(setupSourceSinkTopology(), config);
+
+        testDriver.pipeInput(consumerRecord1);
+        testDriver.pipeInput(consumerRecord1);
+
+        final Iterator<ProducerRecord<byte[], byte[]>> output = testDriver.iterateOutput(SINK_TOPIC_1).iterator();
+
+        final ProducerRecord outputRecord1 = output.next();
+
+        assertEquals(key1, outputRecord1.key());
+        assertEquals(value1, outputRecord1.value());
+        assertEquals(SINK_TOPIC_1, outputRecord1.topic());
+
+        final ProducerRecord outputRecord2 = output.next();
+
+        assertEquals(key1, outputRecord2.key());
+        assertEquals(value1, outputRecord2.value());
+        assertEquals(SINK_TOPIC_1, outputRecord2.topic());
+
+    }
+
+    @Test
+    public void shouldReturnIteratorForMultipleRecordsWithSerDes() {
+        testDriver = new TopologyTestDriver(setupSourceSinkTopology(), config);
+
+        testDriver.pipeInput(recordFactory.create(SOURCE_TOPIC_1, "Key1", 12345L));
+        testDriver.pipeInput(recordFactory.create(SOURCE_TOPIC_1, "Key2", 6789L));
+
+        final Iterator<ProducerRecord<String, Long>> output = testDriver.iterateOutput(SINK_TOPIC_1, stringDeserializer, longDeserializer).iterator();
+
+        final ProducerRecord outputRecord1 = output.next();
+
+        assertEquals("Key1", outputRecord1.key());
+        assertEquals(12345L, outputRecord1.value());
+        assertEquals(SINK_TOPIC_1, outputRecord1.topic());
+
+        final ProducerRecord outputRecord2 = output.next();
+
+        assertEquals("Key2", outputRecord2.key());
+        assertEquals(6789L, outputRecord2.value());
+        assertEquals(SINK_TOPIC_1, outputRecord2.topic());
+
+    }
+
+    @Test
+    public void shouldReturnEmptyListForIterateOutput() {
+        testDriver = new TopologyTestDriver(setupSourceSinkTopology(), config);
+
+        assertFalse(testDriver.iterateOutput(SINK_TOPIC_1).iterator().hasNext());
+
+    }
+
+    @Test
+    public void shouldReturnEmptyListForIterateOutputWithSerDes() {
+        testDriver = new TopologyTestDriver(setupSourceSinkTopology(), config);
+
+        assertFalse(testDriver.iterateOutput(SINK_TOPIC_1, stringDeserializer, longDeserializer).iterator().hasNext());
+
```

Comment:
nit: remove empty line

Reply:
@cadonna Hope I have addressed all your findings in the latest commit.